### PR TITLE
perf(hash): Use new hash map

### LINKED ISSUE #62: Use a faster `HashMap` crate
I don't think hash collision is a concern for this crate. Consider using a faster `HashMap` impl than `std::collections::HashMap`.

Reference:
https://nnethercote.github.io/perf-book/hashing.html
https://nnethercote.github.io/perf-book/linting.html#disallowing-types

### PR DESCRIPTION
**Pull request by malik672**
implement a faster hashmap
bump to 1.0.0

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "uniswap-sdk-core"
-version = "0.23.0"
+version = "1.0.0"
```
Comment (shuhuiluo) [inline Cargo.toml]: Let's not jump to 1.0.0 rn. I will have a PR updating dependencies.

**Review by shuhuiluo — CHANGES_REQUESTED** [review]
Please fix linting.

File: Cargo.toml (hunk 1 above)
Comment (malik672) [reply Cargo.toml]: ok

**Review by shuhuiluo — CHANGES_REQUESTED** [review]
Remove `missing_docs` attribute to fix linting.

File: Cargo.toml (hunk 1 above)
Comment (shuhuiluo) [inline Cargo.toml]:
```suggestion
version = "0.24.0"
```

**Review by shuhuiluo — CHANGES_REQUESTED** [review]

**Review by shuhuiluo — APPROVED** [review]

**Comment by shuhuiluo** [comment]
Please use squash merge to keep the main branch clean.

**Pull request merged**